### PR TITLE
Added missing @Deprecated annotations

### PR DESCRIPTION
**Pull request by tbw777**
These annotations helpfully to prevent usage of deprecated API. For example: developer can see crossed out method/field name if this item is deprecated.

### CODE REVIEW

**Comment by mbien** [comment]
@neilcsmith-net this one might be a candidate for NB 17, what do you think?

**Comment by neilcsmith-net** [comment]
@mbien in principle should be safe.  Will also add into the sig files / tests.  Your call!

If we're sure code cleanup changes have no behavioural changes (some of these PR do) then probably safe to merge even this close to branch.  Others I would prefer we push until after NB17 is released - some could cause some "fun" with sync PRs?

**Comment by mbien** [comment]
@neilcsmith-net ok i haven't considered sig tests causing trouble - good that you mention that. I was only thinking about this PR here with the deprecation annotations - I just removed all other cleanup PRs from milestone 17.

Also.. the sig test seems to throw `[sigtest] java.lang.ClassFormatError: Index out of the constant pool bounds` which it probably shouldn't do.

**Comment by mbien** [comment]
@ebarboni you did recently a lot of doc related work and know also a lot more about the NB build than me. You think adding the deprecation annotation would cause any issues anywhere? I think this would be an important PR to be merged.

**Comment by mbien** [comment]
@matthiasblaesing do you see any potential issues caused by bulk adding the deprecated annotation for already in-doc deprecated items? (beside more compiler warnings)

I can't come up with a good reason to not merge this for NB17.

File: ide/xml.text/src/org/netbeans/modules/xml/text/completion/ElementResultItem.java
Comment (matthiasblaesing) [inline ide/xml.text/src/org/netbeans/modules/xml/text/completion/ElementResultItem.java:128]: This is wrong. The @deprecated javadoc annotation was on the `EndTag` class, nut on getPaintColor. The author commenting out the class should have removed it.

File: java/java.source.base/src/org/netbeans/modules/java/source/pretty/VeryPretty.java
Comment (matthiasblaesing) [inline java/java.source.base/src/org/netbeans/modules/java/source/pretty/VeryPretty.java:2226]: This is wrong. The `@deprecated` string in the javadoc is part of a list of tags.

File: java/javadoc/src/org/netbeans/modules/javadoc/hints/AddTagFix.java
Comment (matthiasblaesing) [inline java/javadoc/src/org/netbeans/modules/javadoc/hints/AddTagFix.java:147]: This is wrong. The @deprecated string in the javadoc is part of a list of tags.

File: java/refactoring.java/src/org/netbeans/modules/refactoring/java/plugins/ChangeParamsTransformer.java
Comment (matthiasblaesing) [inline java/refactoring.java/src/org/netbeans/modules/refactoring/java/plugins/ChangeParamsTransformer.java:918]: This is wrong. The @deprecated string in the javadoc is part of a list of tags.

**Review by matthiasblaesing — CHANGES_REQUESTED** [review]
In general ok, but there are problems (see inline comments).

For the point Neil made, this is the result (example for `editor.mimelookup`):

```diff
# This patch file was generated by NetBeans IDE
# It uses platform neutral UTF-8 encoding and \n newlines.
--- a/platform/editor.mimelookup/nbproject/org-netbeans-modules-editor-mimelookup.sig
+++ b/platform/editor.mimelookup/nbproject/org-netbeans-modules-editor-mimelookup.sig
@@ -1,5 +1,5 @@
 #Signature file v4.1
-#Version 1.58
+#Version 1.59
 
 CLSS public abstract interface !annotation java.lang.Deprecated
  anno 0 java.lang.annotation.Documented()
@@ -49,9 +49,13 @@
 
 CLSS public final org.netbeans.api.editor.mimelookup.MimeLookup
 meth public <%0 extends java.lang.Object> org.openide.util.Lookup$Result<{%%0}> lookup(org.openide.util.Lookup$Template<{%%0}>)
+ anno 0 java.lang.Deprecated()
 meth public <%0 extends java.lang.Object> {%%0} lookup(java.lang.Class<{%%0}>)
+ anno 0 java.lang.Deprecated()
 meth public org.netbeans.api.editor.mimelookup.MimeLookup childLookup(java.lang.String)
+ anno 0 java.lang.Deprecated()
 meth public static org.netbeans.api.editor.mimelookup.MimeLookup getMimeLookup(java.lang.String)
+ anno 0 java.lang.Deprecated()
 meth public static org.openide.util.Lookup getLookup(java.lang.String)
 meth public static org.openide.util.Lookup getLookup(org.netbeans.api.editor.mimelookup.MimePath)
 supr org.openide.util.Lookup
```

The anntoation is recorded in the signature, but I don't see a problem with that. Even when branching that should not be an issue. The additional `@Deprecated` annotations might cause more merge conflicts, but that would be true for any large scale cleanup. The size here seems doable.

**Comment by tbw777** [comment]
fixed

File: java/refactoring.java/src/org/netbeans/modules/refactoring/java/plugins/ChangeParamsTransformer.java
Comment (mbien) [inline java/refactoring.java/src/org/netbeans/modules/refactoring/java/plugins/ChangeParamsTransformer.java:919]: you made it worse. This is *not* deprecated. The annotation is just mentioned in the javadoc in a list.

Comment (tbw777) [reply java/refactoring.java/src/org/netbeans/modules/refactoring/java/plugins/ChangeParamsTransformer.java]: fixed

File: enterprise/j2ee.sun.ddui/src/org/netbeans/modules/j2ee/sun/share/configbean/SunONEDeploymentConfiguration.java
Comment (mbien) [inline enterprise/j2ee.sun.ddui/src/org/netbeans/modules/j2ee/sun/share/configbean/SunONEDeploymentConfiguration.java:132]: one space is missing. 4 spaces per indent.

**Review by mbien — COMMENTED** [review]
marked one formatting adjustment. After that please squash everything to one commit.

If you need instructions how to squash please say so I can help.

**Review by mbien — APPROVED** [review]
thanks. looked through it again and it looks good to me.

**Review by matthiasblaesing — APPROVED** [review]
Looks sane to me.

**Pull request merged**
